samcli/lib/build/app_builder.py

```diff
@@ -184,6 +188,13 @@ def build(self) -> ApplicationBuildResult:
             Returns the build graph and the path to where each resource was built as a map of resource's LogicalId
             to the path string
         """
+        # Remove excluded resources
```

Comment:
Why do we need to remove excluded resources here and overwrite `self._resources_to_build`?
Doesn't `self._resources_to_build` already contain only the non-excluded resources when instantiating the `ApplicationBuilder` instance?

Reply:
Initially I thought the parameters were being altered in some way behind the scenes, but further testing has shown me this is not the case, and this chunk isn't needed after all. I will revise this in my next commit.